module/zfs/abd.c

```diff
@@ -215,15 +221,37 @@ abd_fini(void)
 		kstat_delete(abd_ksp);
 		abd_ksp = NULL;
 	}
+}
+
+#else
 
-	kmem_cache_destroy(abd_chunk_cache);
-	abd_chunk_cache = NULL;
+struct page;
+#define	kpm_enable			1
+#define	abd_alloc_chunk() \
+	((struct page *)kmem_alloc(PAGESIZE, KM_SLEEP))
+#define	abd_free_chunk(chunk) 		kmem_free(chunk, PAGESIZE)
```

Comment:
Potential `PAGE_SHIFT` discrepancies will be resolved by dkms. I was thrown off by a colleague's driver debugging story with two arm64 systems with different page sizes.